src/unistd.rs

```diff
@@ -114,8 +114,13 @@ pub fn chdir<P: ?Sized + NixPath>(path: &P) -> Result<()> {
 #[inline]
 pub fn chown<P: ?Sized + NixPath>(path: &P, owner: Option<uid_t>, group: Option<gid_t>) -> Result<()> {
     let res = try!(path.with_nix_path(|cstr| {
-        // We use `0 - 1` to get `-1 : {u,g}id_t` which is specified as the no-op value for chown(3).
-        unsafe { libc::chown(cstr.as_ptr(), owner.unwrap_or(0 - 1), group.unwrap_or(0 - 1)) }
+        // According to the POSIX specification, -1 is used to indicate that
+        // owner and group, respectively, are not to be changed. Since uid_t and
+        // gid_t are unsigned types, we use std::num::Wrapping to get '-1'.
+        use std::num::Wrapping;
+        unsafe { libc::chown(cstr.as_ptr(),
+                             owner.unwrap_or((Wrapping(0 as uid_t) - Wrapping(1 as uid_t)).0),
+                             group.unwrap_or((Wrapping(0 as gid_t) - Wrapping(1 as gid_t)).0)) }
```

Comment:
one last nit, can we use `0.wrapping_sub(1)` instead of using `Wrapping`? I think it'll be easier to read and doesn't require an extra import.